### PR TITLE
Update xml langlib methods

### LINKED ISSUE #22509: Update xml langlib
**Description:**
Update the xml langlib to align with https://github.com/ballerina-platform/ballerina-spec/issues/480

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by irshadnilam**
## Purpose
$title
Modifies `elements()` method to support querying elements by name
Adds `elementChildren()` method
Adds `children()` method

Fixes #22509

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java [hunk 1]
```diff
@@ -15,49 +15,50 @@
...
  * Return lift getChildren over sequences.
  *
  * @since 1.2
  */
 @BallerinaFunction(
-        orgName = "ballerina", packageName = "lang.__internal",
-        functionName = "children",
+        orgName = "ballerina", packageName = "lang.xml",
+        functionName = "elementChildren",
         args = {@Argument(name = "xmlValue", type = TypeKind.XML)},
```
Comment (rdhananjaya) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java:38]: Need to add `name` parameter here.

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java [hunk 2]
```diff
@@ -15,49 +15,50 @@
...
         returnType = {@ReturnType(type = TypeKind.XML)},
         isPublic = true
 )
-public class Children {
+public class ElementChildren {
 
-    public static XMLValue children(Strand strand, XMLValue xmlVal) {
+    public static XMLValue elementChildren(Strand strand, XMLValue xmlVal, Object name) {
```
Comment (rdhananjaya) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java:44]: I think we can simply do a `name == null` check here.

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java [hunk 3]
```diff
@@ -15,49 +15,50 @@
...
+                return (XMLValue) ((XMLItem) xmlVal).children().elements((String) name);
+            }
+            return (XMLValue) ((XMLItem) xmlVal).children().elements();
         } else if (xmlVal.getNodeType() == XMLNodeType.SEQUENCE) {
-            ArrayList<BXML> liftedChildren = new ArrayList<>();
-            XMLSequence sequence = (XMLSequence) xmlVal.elements();
-            for (BXML bxml : sequence.getChildrenList()) {
-                liftedChildren.addAll(((XMLItem) bxml).getChildrenSeq().getChildrenList());
+            if (namedQuery) {
+                 return  ((XMLSequence) xmlVal).elements((String) name);
```
Comment (rdhananjaya) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java:53]: We need to lift children of children in a sequence 
Refer the comment in langlib signature `# This is equivalent to `children(x).elements(nm)`.`

File: langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal [hunk 4]
```diff
@@ -217,3 +217,52 @@ function testGet() returns [xml|error, xml|error, xml|error, xml|error, xml|erro
 
     return [e1, e2, c1, item, item2];
 }
+
+function testChildren() returns [[int, string, string, string, string], [int, string, string, string]] {
```
Comment (rdhananjaya) [inline langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal:221]: Shall we move these test cases into Ballerina, as we are targetting to move away from testing in java classes.
You can use string comparisons and also comparisons with xml literals.

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java (hunk 1 above)
Comment (irshadnilam) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java]: Added

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java (hunk 2 above)
Comment (irshadnilam) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java:48]: +1

File: langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal (hunk 4 above)
Comment (irshadnilam) [reply langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal]: +1, fixed.

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java [hunk 5]
```diff
@@ -29,35 +29,54 @@
...
  *
  * @since 1.2
  */
 @BallerinaFunction(
-        orgName = "ballerina", packageName = "lang.__internal",
-        functionName = "children",
-        args = {@Argument(name = "xmlValue", type = TypeKind.XML)},
+        orgName = "ballerina", packageName = "lang.xml",
+        functionName = "elementChildren",
+        args = {@Argument(name = "xmlValue", type = TypeKind.XML), @Argument(name = "name", type = TypeKind.STRING)},
```
Comment (rdhananjaya) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/ElementChildren.java:42]: Shall we use the same param names as in .bal file
And also type of parameter `name` is not String it's a union of string and nil

File: langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal [hunk 6]
```diff
@@ -217,3 +217,73 @@ function testGet() returns [xml|error, xml|error, xml|error, xml|error, xml|erro
...
+function assert(anydata actual, anydata expected) {
+    if (expected != actual) {
+        typedesc<anydata> expT = typeof expected;
+        typedesc<anydata> actT = typeof actual;
+        string reason = "expected [" + expected.toString() + "] of type [" + expT.toString()
+                            + "], but found [" + actual.toString() + "] of type [" + actT.toString() + "]";
+        error e = error(reason);
+        panic e;
+    }
+}
```
Comment (rdhananjaya) [inline langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal:289]: Shall we add a few test cases with namespaces? You can refer `XMLAccessTest` in jballerina-unit-test module.

Comment (irshadnilam) [reply langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal:328]: Added with https://github.com/ballerina-platform/ballerina-lang/pull/22621/commits/b11efbfc5a16c327b1e4a7cc96c276d3141f910a

**Pull request merged**